specs/ProcessInfo.md

```diff
@@ -0,0 +1,269 @@
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_GPU);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_PPAPI_PLUGIN);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_PPAPI_BROKER);
+        KIND_ENTRY(COREWEBVIEW2_PROCESS_KIND_UNKNOWN);
+
+#undef KIND_ENTRY
+    }
+
+    return L"PROCESS KIND: " + std::to_wstring(static_cast<uint32_t>(kind));
+}
+
+// Get the process info
+//! [ProcessInfoChanged]
+void ProcessComponent::PerformanceInfo()
+{
+    std::wstring result;
+    UINT processListCount;
+    CHECK_FAILURE(m_processCollection->get_Count(&processListCount));
+
+    if (processListCount == 0)
+    {
+        result += L"No process found.";
+    }
+    else
+    {
+        result += std::to_wstring(processListCount) + L" process(s) found";
+        result += L"\n\n";
+        for (UINT i = 0; i < processListCount; ++i)
+        {
+            wil::com_ptr<ICoreWebView2StagingProcessInfo> processInfo;
+            CHECK_FAILURE(m_processCollection->GetValueAtIndex(i, &processInfo));
+
+            UINT32 processId = 0;
+            COREWEBVIEW2_PROCESS_KIND kind;
+            CHECK_FAILURE(processInfo->get_Id(&processId));
+            CHECK_FAILURE(processInfo->get_Kind(&kind));
+
+            WCHAR id[4096] = L"";
+            StringCchPrintf(id, ARRAYSIZE(id), L"Process ID: %u", processId);
+
+            HANDLE processHandle = OpenProcess(PROCESS_ALL_ACCESS, FALSE, processId);
+            PROCESS_MEMORY_COUNTERS_EX pmc;
+            GetProcessMemoryInfo(
+                processHandle, reinterpret_cast<PROCESS_MEMORY_COUNTERS*>(&pmc), sizeof(pmc));
+            SIZE_T virtualMemUsed = pmc.PrivateUsage / 1024;
+            WCHAR memory[4096] = L"";
+            StringCchPrintf(memory, ARRAYSIZE(memory), L"Memory: %u", virtualMemUsed);
+
+            result = result + id + L" | Process Kind: " + ProcessKindToString(kind) + L" | " +
+                     memory + L" KB\n";
+        }
+    }
+    MessageBox(nullptr, result.c_str(), L"Memory Usage", MB_OK);
+}
+//! [ProcessInfoChanged]
+```
+
+# API Details    
+```
+interface ICoreWebView2Environment6;
+interface ICoreWebView2ProcessInfo;
+interface ICoreWebView2ProcessInfoCollection;
+interface ICoreWebView2ProcessInfoChangedEventHandler;
+
+[v1_enum]
+typedef enum COREWEBVIEW2_PROCESS_KIND {
+  /// Indicates the browser process kind.
+  COREWEBVIEW2_PROCESS_KIND_BROWSER,
+
+  /// Indicates the render process kind.
+  COREWEBVIEW2_PROCESS_KIND_RENDERER,
+
+  /// Indicates the utility process kind.
+  COREWEBVIEW2_PROCESS_KIND_UTILITY,
+
+  /// Indicates the sandbox helper process kind.
+  COREWEBVIEW2_PROCESS_KIND_SANDBOX_HELPER,
+
+  /// Indicates the GPU process kind.
+  COREWEBVIEW2_PROCESS_KIND_GPU,
+
+  /// Indicates the PPAPI plugin process kind.
+  COREWEBVIEW2_PROCESS_KIND_PPAPI_PLUGIN,
+
+  /// Indicates the PPAPI plugin broker process kind.
+  COREWEBVIEW2_PROCESS_KIND_PPAPI_BROKER,
+
+  /// Indicates the process of unspecified kind.
+  COREWEBVIEW2_PROCESS_KIND_UNKNOWN,
```

Comment:
I think we should just omit this member. Future process types can just return new values. Avoids needing to quirk based on caller expected version.

For example, suppose we later add two new types of helper process, the "aroma manager" and the "aura cleanser". Do we add these as two new kinds `COREWEBVIEW2_PROCESS_KIND_AROMA` and `COREWEBVIEW2_PROCESS_KIND_AURA`? Or do we report both of them as `COREWEBVIEW2_PROCESS_KIND_UNKNOWN`?

If we report them as new `KIND`s, then we would have to add a quirk so that older clients receive both of them as  `COREWEBVIEW2_PROCESS_KIND_UNKNOWN`.

I think better is to simply not have an "unknown" value. If we later define new kinds, we just expand the list.

Existing clients will pass the "kind" through a switch statement, it won't match anything, and the `default` will classify it as "unknown" anyway.